engine/engine.go

```diff
@@ -117,35 +129,44 @@ func (c *TofuEngine) Run(req *tgengine.RunRequest, stream tgengine.Engine_RunSer
 	// Stream stderr
 	go func() {
 		defer wg.Done()
+
 		reader := transform.NewReader(stderrPipe, unicode.UTF8.NewDecoder())
 		bufReader := bufio.NewReader(reader)
+
 		for {
 			char, _, err := bufReader.ReadRune()
 			if err != nil {
-				if err != io.EOF {
+				if !errors.Is(err, io.EOF) {
 					log.Errorf("Error reading stderr: %v", err)
 				}
+
 				break
 			}
+
 			if err = stream.Send(&tgengine.RunResponse{Stderr: string(char)}); err != nil {
 				log.Errorf("Error sending stderr: %v", err)
 				return
 			}
 		}
 	}()
 	wg.Wait()
-	err = cmd.Wait()
+
 	resultCode := 0
+
+	err = cmd.Wait()
 	if err != nil {
```

Comment:
```suggestion
	if err := cmd.Wait(); err != nil {
```

Since we don't need this error anywhere else, it's better to hide its scope. And the readability of the code improves.